### ISSUE TITLE
Rendering issue with ArtemisLib and Crawling Anim

### ISSUE THREAD

**Issue by EdoEquin0x** [Fuzss/fivefeetsmall#137]
- Install ArtemisLib and AquaAcro and turn on the ArtemisLib Integration
- Use a mod such as GulliverReloaded
- Crawl in front of another player with a size scaling of 1 (normal size) or less, this happen:
 
![2024-04-08_12 28 29](https://github.com/embeddedt/aquaacrobatics/assets/69121197/345bd0e1-eb40-467a-9832-e72d6a8e2605)
![2024-04-08_12 28 23](https://github.com/embeddedt/aquaacrobatics/assets/69121197/fc103080-007b-48c7-833f-0433cc935d42)

- Set the size scaling of the crawling player to at least 1.2
- Crawl again, it's working: 
![2024-04-08_12 28 36](https://github.com/embeddedt/aquaacrobatics/assets/69121197/11e88ac8-490e-4a3a-807b-9fabc6cfd5de)

So basically it seems that if the player is of a size scaling smaller than 1.2, the crawling animation render is not working properly.

**Comment by EdoEquin0x** [comment]
Some other screens, I think the issue is coming from 

> GlStateManager.translate(evt.getX() / widthAttribute - evt.getX(), evt.getY() / heightAttribute - evt.getY(), evt.getZ() / widthAttribute - evt.getZ());

located in the onEntityRenderPre method in the AttachAttributesFix class

![2024-04-28_20 03 36](https://github.com/embeddedt/aquaacrobatics/assets/69121197/7f3ee230-796d-44eb-ba5b-25d23ba69a70)
![2024-04-28_20 03 39](https://github.com/embeddedt/aquaacrobatics/assets/69121197/33e9f1f3-cd33-4dfb-8934-1508a90edee2)

**Comment by EdoEquin0x** [comment]
If that can help you: Did a partial fix on my part by simply reducing the hitbox of the player by 60% if the player is crouching (instead of scaling the render). Works like a charm so far

**Comment by xef5000** [comment]
Noticed a similar issue on my end (forge 1.12.2).
I used:
- AquaAcrobatics https://www.curseforge.com/minecraft/mc-mods/aqua-acrobatics
- Gulliver Reborn https://www.curseforge.com/minecraft/mc-mods/gulliver-reborn
- Artemis Lib https://www.curseforge.com/minecraft/mc-mods/artemislib
When swimming on a server, the player became invisible. Thankfully, we were able to remove gulliver reborn and it mostly works (still some issues, rarely) as it wasn't an essential part of our infrastructure (we use ArtemisLib in our own mod to modify player size in different scenarios).
But yeah, it is quite annoying.

I'm guessing it is caused by some capability syncing issues. I could be wrong though!
So yeah, problem comes only from gulliver reborn, not artemis

**Comment by EdoEquin0x** [comment]
May you tell me more about those scenarios you use your homemade mode for ?

**Comment by xef5000** [comment]
So I'm a lead dev at a modded minecraft server.
We have our own mod that we make in-house along with many other mods from open-source projects (like aquaacrobatics).
We have +-50 mods in total, including gulliver, artemis and aquaacrobatics.
When two players are on the same world and a player starts swimming in front of the other, the swiming player becomes 100% invisible to the observing player. Particles still happen though.
We weren't using any size modifiers. We just had the mod installed.
